ui/apps/platform/src/Containers/Collections/CollectionFormPage.tsx

```diff
@@ -1,16 +1,90 @@
-import React from 'react';
+import React, { CSSProperties, useCallback } from 'react';
+import {
+    Breadcrumb,
+    BreadcrumbItem,
+    Button,
+    Divider,
+    PageSection,
+    Title,
+} from '@patternfly/react-core';
+
+import useRestQuery from 'Containers/Dashboard/hooks/useRestQuery';
+import { getCollection } from 'services/CollectionsService';
+import useLayoutSpaceObserver from 'hooks/useLayoutSpaceObserver';
+import BreadcrumbItemLink from 'Components/BreadcrumbItemLink';
+import { collectionsBasePath } from 'routePaths';
 import { CollectionPageAction } from './collections.utils';
+import CollectionForm from './CollectionForm';
 
 export type CollectionsFormPageProps = {
     pageAction: CollectionPageAction;
 };
 
+const noopRequest = {
+    request: Promise.resolve(undefined),
+    cancel: () => {},
+};
+
+function styleVarsForResultsList(resultListSpaceOffset: number): CSSProperties {
+    return {
+        '--collection-results-container-max-height': `calc(
+            100vh -
+            var(--pf-c-page__header--MinHeight) -
+            var(--pf-c-page__main-section--PaddingTop) -
+            var(--pf-c-page__main-section--PaddingBottom) -
+            ${resultListSpaceOffset}px
+        )`,
+    } as CSSProperties;
+}
+
+const observedClass = 'collections-observe-layout';
+
 function CollectionsFormPage({ pageAction }: CollectionsFormPageProps) {
-    return pageAction.type === 'create' ? (
-        <>{pageAction.type}</>
-    ) : (
+    const action = pageAction.type;
+    const collectionId = action !== 'create' ? pageAction.collectionId : undefined;
+    const collectionFetcher = useCallback(() => {
+        return collectionId ? getCollection(collectionId) : noopRequest;
+    }, [collectionId]);
+
+    const { data } = useRestQuery(collectionFetcher);
+
+    const watchElements = Array.from(document.getElementsByClassName(observedClass));
```

Comment:
The `useLayoutSpaceObserver` hook and the `IntersectionObserver` inside it require a reference to DOM nodes in order to track the space usage in the container. There are three main approaches I looked at when setting this up:

1. Use the imperative `document.getElementsByClassName` method to access the DOM directly.
This is probably the least React-idiomatic, but most straightforward, way to do this and it what I have now. This simply queries the DOM for a class that is defined in this component to get a reference to the correct elements. The `undefined` values on the first render do not matter, since the `useEffect` in the hook triggers an update when the DOM elements are mounted.

2. Use `Refs` part one. Ideally I'd like to get a ref to the `<section>` in the PF `<PageSection>` components, but since those are React function components, the only way to do so is via `forwardRef`. Unfortunately, PF doesn't use `forwardRef` for this component so this option is out.

3. Use `Refs` part two. For this use case, I could wrap the sections I need reference to with a simple `<div ref={myRef}>` and the functionality should be identical. The downside here is that it would make the markup a bit clunky, as well as introduce CSS conflicts that would need to be overridden in order to display the PF components correctly. (We have a global CSS rule that messes up styles if the immediate descendant of a `<Page>` is not a `<PageSection>`.)